docs/advanced/cloudinitcrd.md

```diff
@@ -0,0 +1,144 @@
+title: "CloudInit CRD"
+---
+
+<head>
+  <link rel="canonical" href="https://docs.harvesterhci.io/v1.1/advanced/cloudinitcrd"/>
+</head>
+
+_Available as of v1.3.0_
+
+Harvester now offers the capability to configure Harvester OS specific settings via the `CloudInit` CRD.
+
+This can be used to configure underlying Harvester OS specific options manually or via Gitops
+
+## Background
+Harvester OS uses the [elemental-toolkit](https://rancher.github.io/elemental-toolkit/docs/reference/cloud_init/) which ships with its unique cloud-init support. 
+
+As part of the Harvester installation, settings defined by users are written to an elemental cloud-init file in /oem directory. As Harvester OS itself is immutable, the cloud-init file ensures node specific settings are applied on each reboot. 
+
+The Harvester `CloudInit` CRD exposes the `elemental cloud-init` via a Kubernetes CRD. This allows users to perform additional node specific changes post Harvester install without having to follow the process of making the rootfs writable.
+
+The `CloudInit` CRD is persisted and sync across to the underlying hosts via Harvester, which ensures changes made directly to OS are not lost across reboots or upgrades.
+
+:::note
+
+`CloudInit CRD` is a cluster scoped resource. Please ensure you have appropriate access via Rancher rbac to access the resource.
+
+:::
+
+## Getting started
+
+The following example adds additional ssh keys to all nodes in an existing Harvester cluster.
+
+```yaml
+apiVersion: node.harvesterhci.io/v1beta1
+kind: CloudInit
+metadata:
+  name: my-ssh-keys
+spec:
+  matchSelector: 
+    harvesterhci.io/managed: "true"
+  filename: 99-my-ssh-keys
+  contents: |
+    stages:
+      network:
+        - name: "add my ssh keys"
+          authorized_keys:
+            rancher:
+            - ssh-rsa key1
+            - ssh-rsa key2
+  paused: false
+```
+
+Details of the CRD spec are as follows:
+
+* `matchSelector (required)`: defines label selector to filter nodes the change needs to be applied to. `harvesterhci.io/managed: "true"` is a harvester specific label, akin to select all nodes
+* `filename (required)`: defines the name of file in /oem. cloud-init files in /oem are applied in alphabetical order, and this can be used to ensure how file changes are applied during boot
+* `content (required)`: defines the inline content for the elemental cloud-init resource which will be written to target nodes
+* `paused (optional)`: optional field to pause the reconcile of `CloudInit` CRD. A elemental cloud-init file being managed by the `CloudInit` CRD is watched by the Harvester controllers. Any direct changes made to these files are immediately reconcilled back to the defined state, unless the CRD is paused. 
+
+
+Once the object is created, the user can login to the target nodes to verify the results.
+
+In our case, a file named `/oem/99-my-ssh-keys.yaml` is created and subsequently monitored by the Harvester controllers.
```

Comment:
```suggestion
The `spec` field contains the following:

- `matchSelector (required)`: Label selector used to identify the nodes that the change must be applied to. `harvesterhci.io/managed: "true"` is a Harvester-specific label that you can use to select all nodes.
- `filename (required)`: Name of the file in `/oem`. cloud-init files in `/oem` are applied in alphabetical order. This can be used to ensure that file changes are applied during booting.
- `content (required)`: Inline content for the Elemental cloud-init resource that is written to target nodes.
- `paused (optional)`: Used to pause `CloudInit` CRD reconciliation. The Harvester controllers monitor Elemental cloud-init files that are managed by the `CloudInit` CRD. Direct changes made to these files are immediately reconciled back to the defined state unless the CRD is paused. 

Once the object is created, you can log in to the target nodes to verify the results.

In the following example, a file named `/oem/99-my-ssh-keys.yaml` is created and subsequently monitored by the Harvester controllers.
```